src/main/java/uk/gov/pay/api/filter/LoggingFilter.java

```diff
@@ -1,34 +1,52 @@
 package uk.gov.pay.api.filter;
 
+import com.google.common.base.Stopwatch;
+import org.jboss.logging.MDC;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.servlet.*;
 import javax.servlet.http.HttpServletRequest;
 import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
 
 public class LoggingFilter implements Filter {
 
+    public static final String HEADER_REQUEST_ID = "X-Request-Id";
     private static final Logger logger = LoggerFactory.getLogger(LoggingFilter.class);
 
     @Override
-    public void init(FilterConfig filterConfig) throws ServletException {}
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
 
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
         String requestURL = ((HttpServletRequest) servletRequest).getRequestURI();
+        String requestMethod = ((HttpServletRequest) servletRequest).getMethod();
+        Optional<String> requestIdMaybe = Optional.ofNullable(((HttpServletRequest) servletRequest).getHeader(HEADER_REQUEST_ID));
+        String requestId = requestIdMaybe.orElse("");
```

Comment:
Lemme suggest a shortcut here, not that vital :) instead of lines 31-33 you could write

> requestId =StringUtils.defaultString(((HttpServletRequest)servletRequest).getHeader(HEADER_REQUEST_ID));